### DISCUSSION TITLE
Process response from API request in Action

### DISCUSSION THREAD

**Discussion by micebrain** [Budibase/budibase#3318] [Help] [answered]
Hi,
Thanks for the transformer feature! This works well for transforming response from get request. 
Is there a way to process response from "Action" e.g. as result of "Execute query"?  
The flow is the following:
- form data entry with post request
- server returns a result
- the result can be processed and stored or displayed  
I am thinking of the ability to display more specific error messages from server.

**Comment by mjashanks** [comment, marked as answer]
At the moment, the only way to use data in the app is through a Data Provider.

You might be able to half-achieve this using an automation. It'd wouldn't update immediately though, I think a manual refresh would be needed.

So, you could try...
1. Creating a Budibase table, called "Form Posts"
2. Setup all the same column in this table, as are in your form
3. Have your form `Save Row` to this table
4. Create a new automation, with a "Create Row" trigger on your "Form Posts" table
5. Add a step to "Execute Query" - this sends to your API
6. Add a second step to Update a column your "Form posts" row, with your POST response

Automations all happen in the background, so you won't be able to display a result immediately on-screen. But once the automation has been completed, you'll be able to see the result in the record that was created.

It's not ideal, but you might be able to get something useful this way.

**Reply by micebrain** [reply]
Thanks for the hint. It can potentially work. However is there any plan to have such feature in future releases?

**Reply by rpmcoding** [reply]
I too wound very much like this feature

**Comment by nntrongnghia** [comment]
@micebrain Hi, thank you for such a great product.
I also have the same issue where I need to know the response from the REST API server. Do you know how to achieve this or will this feature be included in the future releases?

**Reply by rpmcoding** [reply]
Yes I need this to.  Perhaps just allowing a response to be saved in the session.

**Reply by alejoar** [reply]
+1 please allow the response (content, status code, etc) to be saved in the state
